dbos/_dbos.py

```diff
@@ -1304,6 +1314,155 @@ def set_authentication(
         ctx.authenticated_user = authenticated_user
         ctx.authenticated_roles = authenticated_roles
 
+    @classmethod
+    def write_stream(cls, key: str, value: Any) -> None:
+        """
+        Write a value to a stream.
+
+        Args:
+            key(str): The stream key / name within the workflow
+            value(Any): A serializable value to write to the stream
+
+        """
+        cur_ctx = get_local_dbos_context()
+        if cur_ctx is None or not cur_ctx.is_within_workflow():
+            raise DBOSException("write_stream() must be called from within a workflow")
+
+        def fn() -> None:
+            ctx = assert_current_dbos_context()
+            _get_dbos_instance()._sys_db.write_stream(ctx.workflow_id, key, value)
+
+        return _get_dbos_instance()._sys_db.call_function_as_step(
```

Comment:
Question: when we do workflow fork, do we also fork workflow events and streams? If not, then the client working with the new workflow can get wrong results.

Reply:
Yeah, fork currently doesn't consider messages, events, or streams. Can fix in a future PR.